src/settings/storeconf.cpp

```diff
@@ -9,126 +9,165 @@
-    QFile file(currentPath);
-    if(file.exists())
-        return "conf.json";
-    else
-        return "";
+    getPathToConFile();
+    QFile file(mPathToConFile);
+    if (file.exists())
+    {
+        writeJson("finish");
+    }
+}
+
+void StoreConf::getPathToConFile()
+{
+    mPathToConFile = QDir::currentPath();
+    mPathToConFile += "/conf/";
+    mPathToConFile += mConFile;
+}
+
+void StoreConf::checkConfDir(QString str)
+{
+    if (str == "exe")
+    {
+        QString pathToConDir = QDir::currentPath();
+        pathToConDir += "/conf";
+        if (!QDir(pathToConDir).exists())
+            QDir().mkdir(pathToConDir);
+    }
 }
 
-void StoreConf::writeJson()
+void StoreConf::writeJson(QString mode)
 {
     QJsonObject root_obj;
-    root_obj.insert("organizationName", organizationName);
-    root_obj.insert("applicationVersion", applicationVersion);
-    root_obj.insert("applicationName", applicationName);
-    root_obj.insert("analizerStyle", analizerStyle);
-    root_obj.insert("analizerFontSize", analizerFontSize);
-    root_obj.insert("analizerFontName", analizerFontName);
-    root_obj.insert("cppExtentions", cppExtentions);
+
+    QSettings s;
+    QMap<QString, QString>::const_iterator it = mFields.constBegin();
+    while (it != mFields.constEnd())
+    {
+        if (mode == "start")
+        {                           // copy value from QMap to json
+            root_obj.insert(it.key(), it.value());
+        }
+        else
+        {                           // copy value from QSettings to json
+            if (s.contains(it.key()))
+                root_obj.insert(it.key(), s.value(it.key()).toString());
+        }
+        ++it;
+    }
 
     QJsonDocument json_doc(root_obj);
     QString json_string = json_doc.toJson();
 
-    conFile = conFileDefault;
-
-    QFile save_file(conFile);
+    QFile save_file(mPathToConFile);
     if(!save_file.open(QIODevice::WriteOnly | QIODevice::Truncate)) {
-        writeStatus = false;
+        mWriteStatus = false;
         return;
     } else
-        writeStatus = true;
+        mWriteStatus = true;
 
     save_file.write(json_string.toLocal8Bit());
 }
 
 void StoreConf::readJson()
 {
-    QFile loadFile(conFile);        // close in destructor
+    QFile loadFile(mPathToConFile);                       // closing in it destructor
     if (loadFile.open(QIODevice::ReadOnly))
-        readStatus = true;
-    else {
-        readStatus = false;
+    {
+        mReadStatus = true;
+    }
+    else
+    {
+        mReadStatus = false;
         return;
     }
 
     QByteArray saveData = loadFile.readAll();
-    loadDoc = QJsonDocument::fromJson(saveData);
-    if(loadDoc.isNull() || !loadDoc.isObject() || loadDoc.isEmpty())
-        readStatus = false;
+    mJsonDoc = QJsonDocument::fromJson(saveData);
+    if(mJsonDoc.isNull() || !mJsonDoc.isObject() || mJsonDoc.isEmpty())
+        mReadStatus = false;
     else
-        readStatus = true;
+        mReadStatus = true;
 }
```

Comment:
Value of mReadStatus is overwritten several times without need
There is only one case when it should be assigned true

Reply:
reading of json file can fall in two ways: during  it opening as raw data (line 118) and during it transforming to type QJsonDocument (line 129)
so it checked two times